GoogleForCreators/web-stories-wp
# Redesign: color swatch

## Context

This generalizes the color swatch component used in the color preset panel, the color picker actions, and the color input. Also moves some general functionality to the design system.

## To-do

- [x] Create a new swatch component extract from the editor codebase
- [x] Create storybook story displaying variants
- [x] Add small border on top of button (also for small?)
- [x] Add appropriate focus styling
- [x] Move related util functions to design system
- [x] Replace swatch in color panel
- [x] Replace swatch in color picker actions
- [x] Replace swatch in color input

## User-facing changes
![Screen Shot 2021-03-01 at 21 43 54](https://user-images.githubusercontent.com/637548/109588754-463ea500-7ad7-11eb-933b-d378d91defb0.png)

## Testing Instructions

### QA

This PR can be tested by following these steps:

1. Add a shape to the stage
1. Set the shape color to a solid
1. Add the color to the saved colors panel
1. Observe the color in the saved colors panel and in the color picker
1. Set the color of the shape to a gradient with transparency
1. Observe the gradient in the saved colors panel and in the color picker
1. Select the default background element
1. Observe that the solid color is applicable in the saved colors panel, but the gradient is not (with a tooltip explaining why)
1. Click the pencil in the saved colors to display
1. Observe that the colors are delectable with a cross and all color buttons are enabled
1. Use the tab key to navigate to the first color in the saved colors panel and arrow keys to navigate between the saved colors.
1. Observe the selection focus ring around the currently focused color button.

### UAT

This PR can furthermore be UAT'ed by following these steps:

1. Go to Storybook > Design System > Component > Swatch
2. Observe variants

## Checklist




---

Fixes #6457
